runelite-client/src/main/java/net/runelite/client/plugins/mta/enchantment/EnchantmentRoom.java

```diff
@@ -67,67 +70,47 @@ public void onGameStateChanged(GameStateChanged gameStateChanged)
 		}
 	}
 
-	@Subscribe
-	public void onGameTick(GameTick event)
-	{
-		if (!inside() || !config.enchantment())
-		{
-			return;
-		}
-
-		WorldPoint nearest = findNearestStone();
-		if (nearest != null)
-		{
-			client.setHintArrow(nearest);
-		}
-		else
-		{
-			client.clearHintArrow();
-		}
-	}
-
-	private WorldPoint findNearestStone()
+	private Tile findNearestStone()
 	{
-		WorldPoint nearest = null;
+		Tile nearest = null;
 		double dist = Double.MAX_VALUE;
 		WorldPoint local = client.getLocalPlayer().getWorldLocation();
-		for (WorldPoint worldPoint : dragonstones)
+		for (Tile tile : dragonstones)
 		{
-			double currDist = local.distanceTo(worldPoint);
+			double currDist = local.distanceTo(tile.getWorldLocation());
 			if (nearest == null || currDist < dist)
 			{
 				dist = currDist;
-				nearest = worldPoint;
+				nearest = tile;
 			}
 		}
 		return nearest;
 	}
 
 	@Subscribe
-	public void onItemSpawned(ItemSpawned itemSpawned)
+	public void onItemSpawned(ItemSpawned event)
 	{
-		final TileItem item = itemSpawned.getItem();
-		final Tile tile = itemSpawned.getTile();
-
-		if (item.getId() == ItemID.DRAGONSTONE_6903)
+		if (config.enchantment())
 		{
-			WorldPoint location = tile.getWorldLocation();
-			log.debug("Adding dragonstone at {}", location);
-			dragonstones.add(location);
+			final TileItem item = event.getItem();
+			if (item.getId() == ItemID.DRAGONSTONE_6903) {
+				final Tile tile = event.getTile();
+				log.debug("Adding dragonstone at {}", tile.getWorldLocation());
+				dragonstones.add(tile);
+			}
 		}
 	}
 
 	@Subscribe
-	public void onItemDespawned(ItemDespawned itemDespawned)
+	public void onItemDespawned(ItemDespawned event)
 	{
-		final TileItem item = itemDespawned.getItem();
-		final Tile tile = itemDespawned.getTile();
-
-		if (item.getId() == ItemID.DRAGONSTONE_6903)
-		{
-			WorldPoint location = tile.getWorldLocation();
-			log.debug("Removed dragonstone at {}", location);
-			dragonstones.remove(location);
+		if (config.enchantment()) {

```

Comment:
Why is this added?

Reply:
I added this as a guard since the enchantment room was the only room without them. I modified it to be more consistent with alchemy room (early return).